Rakefile

```diff
@@ -46,10 +48,29 @@ if !Puma.jruby?
     CLEAN.include "lib/puma/puma_http11.rb"
   end
 else
-  # Java (JRuby)
+  task :java => :maven_dependencies
+  # create a task which will download the dependencies if they do not exist already in the working directory
+  task :maven_dependencies do
+    %w(https://repo1.maven.org/maven2/io/netty/netty-common/4.1.47.Final/netty-common-4.1.47.Final.jar
+       https://repo1.maven.org/maven2/io/netty/netty-codec/4.1.47.Final/netty-codec-4.1.47.Final.jar
+       https://repo1.maven.org/maven2/io/netty/netty-buffer/4.1.47.Final/netty-buffer-4.1.47.Final.jar
+       https://repo1.maven.org/maven2/io/netty/netty-handler/4.1.47.Final/netty-handler-4.1.47.Final.jar
+       https://repo1.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.29.Final/netty-tcnative-boringssl-static-2.0.29.Final.jar).each do |dependency|
```

Comment:
netty-tcnative-boringssl-static is a really useful dependency. It makes pulling in boringssl (a fork of OpenSSL) extremely easy.

> This is an uber jar that contains the statically linked libraries for all supported netty-tcnative-boringssl-static-{os_arch}. While this tends to be a fairly large jar, it greatly simplifies the getting started experience since an application need not be concerned with obtaining the correct jar for the platform.

Reply:
I'm confused on why we have to add it?

Reply:
I supply this to _RakeTestTask_ so that the Puma SSL tests on JRuby can easily use OpenSSL without any setup needed for the environment.

The alternative is to dynamically link against OpenSSL on the system but that's more fragile than statically compiling in BoringSSL directly.

NB: BoringSSL is just a slimmed down fork of OpenSSL by Google.